File: check_status.py
import os
from supabase import create_client
from dotenv import load_dotenv

# Load env vars
load_dotenv(".env")
load_dotenv("railway-api/.env.example") # Fallback

url = os.environ.get("SUPABASE_URL") or os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_KEY") or os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not url or not key:
    print("Error: Missing Supabase credentials")
    exit(1)

supabase = create_client(url, key)

# Get the latest analysis
response = supabase.table("analyses").select("*").order("created_at", desc=True).limit(1).execute()

if response.data:
    latest = response.data[0]
    print(f"Latest Analysis:")
    print(f"Key: {latest.get('access_key')}")
    print(f"Channel: {latest.get('channel_name')}")
    print(f"Status: {latest.get('analysis_status')}")
    print(f"Created At: {latest.get('created_at')}")
else:
    print("No analyses found")

File: gap-intel-website/gap-intel-website/check_status.js
const { createClient } = require('@supabase/supabase-js');
// require('dotenv').config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
    console.error('Missing Supabase credentials');
    process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkStatus() {
    const { data, error } = await supabase
        .from('analyses')
        .select('*')
        .eq('access_key', 'GAP-MQ0HKiRM37kP')
        .single();

    if (error) {
        console.error('Error fetching analysis:', error);
        return;
    }

    if (data) {
        console.log('Latest Analysis:');
        console.log(`Key: ${data.access_key}`);
        console.log(`Channel: ${data.channel_name}`);
        console.log(`Email: ${data.email}`);
        console.log(`Status: ${data.analysis_status}`);
        console.log(`Created At: ${data.created_at}`);

        if (data.analysis_result && data.analysis_result.error) {
            console.log(`Error Result: ${data.analysis_result.error}`);
        }
    } else {
        console.log('No analyses found');
    }
}

checkStatus();
